pwndbg/gdblib/proc.py

```diff
@@ -103,6 +105,44 @@ def binary_base_addr(self) -> int:
     def binary_vmmap(self) -> Tuple[pwndbg.lib.memory.Page, ...]:
         return tuple(p for p in pwndbg.gdblib.vmmap.get() if p.objfile == self.exe)
 
+    @pwndbg.lib.memoize.reset_on_start
+    @pwndbg.lib.memoize.reset_on_objfile
+    def dump_elf_data_section(self) -> Optional[Tuple[int, int, bytes]]:
+        """
+        Dump .data section of current process's ELF file
+        """
+        local_path = pwndbg.gdblib.file.get_file(pwndbg.gdblib.proc.exe)
+        with open(local_path, "rb") as f:
+            elffile = ELFFile(f)
+            section = elffile.get_section_by_name(".data")
+            if section:
+                return section["sh_addr"], section["sh_size"], section.data()
+        return None
+
+    @pwndbg.lib.memoize.reset_on_start
+    @pwndbg.lib.memoize.reset_on_objfile
+    def get_data_section_address(self) -> int:
+        """
+        Find .data section address of current process.
+        """
+        out = pwndbg.gdblib.info.files()
+        for line in out.splitlines():
+            if line.endswith(" is .data"):
+                return int(line.split()[0], 16)
+        return 0
+
+    @pwndbg.lib.memoize.reset_on_start
+    @pwndbg.lib.memoize.reset_on_objfile
+    def get_got_section_address(self) -> int:
```

Comment:
We already have something that looks for .got section in current binary in gdblib.elf iirc, do we need this?

Reply:
> We already have something that looks for .got section in current binary in gdblib.elf iirc

Hmm, I didn’t find something related in `gdblib.elf`. Which function did you mean?
Or did you mean this function in `commands.elf`?
https://github.com/pwndbg/pwndbg/blob/f627b614627a58b2229357dc3e9d8d1b048cd738/pwndbg/commands/elf.py#L67-L93

> do we need this?

Since the above approach might need to download the file from the remote server, I'll prefer keeping this function and parsing the output of the `info files`. (but I’m not sure which approach is better tho)